### PR TITLE
Zip

### PR DESCRIPTION
**Pull request by sapirpol**
Add zip builder support

### Checklist
- [X] Code compiles correctly
- [ ] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formating and linting run locally successfully
- [X] All tests passing
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: cmd/module.go [hunk 1]
```diff
@@ -36,6 +43,18 @@ func init() {
 	packModuleCmd.Flags().StringVarP(&packCmdPlatform, "platform", "p", "",
 		"the deployment platform; supported plaforms: cf, xsa, neo")
 
+	// sets the flags of of the command zip module
+	zipModuleCmd.Flags().StringVarP(&zipCmdSrc, "source", "s", "",
```
Comment (allaVolkov) [inline cmd/module.go]: if we provide all these flags we have to add them to configuration's command and fill in buildOpts

File: configs/builder_type_cfg.yaml [hunk 2]
```diff
@@ -12,36 +12,52 @@
...
+  builder-types:
+  - name: maven
+    commands:
+    - command: mvn dependency:copy -Dartifact={{repo-coordinates}} -DoutputDirectory=./
+
+- name: zip
+  info: "the zip builder archives the source folder into a .zip file and saves it as .mtar without making any changes"
+  path: "the zip builder archives the source folder into a .zip file and saves it as .mtar without making any changes"
+  commands:
+  - command: mbt module zip -m={{module-name}} -s={{source}}
```
Comment (allaVolkov) [inline configs/builder_type_cfg.yaml]: all other flags to be provided as well, We have to care of some naming convention for "system" properties

File: internal/artifacts/module_arch.go [hunk 3]
```diff
@@ -32,19 +32,45 @@ func ExecuteBuild(source, target, desc, moduleName, platform string, wdGetter fu
...
+
 	logs.Logger.Infof(`packing the "%v" module...`, moduleName)
 
 	loc, err := dir.Location(source, target, desc, wdGetter)
 	if err != nil {
 		return errors.Wrapf(err, `packing of the "%v" module failed when initializing the location`, moduleName)
 	}
 
-	module, _, err := commands.GetModuleAndCommands(loc, moduleName)
+	module, _, err := commands.GetModuleAndCommands(loc, source, moduleName)
```
Comment (allaVolkov) [inline internal/artifacts/module_arch.go]: please don't use GetModuleAndCommands. if we fail on this it's not good to receive message about commands during normal pack. To receive module info use parse

File: internal/artifacts/module_arch.go [hunk 4]
```diff
@@ -32,19 +32,45 @@ func ExecuteBuild(source, target, desc, moduleName, platform string, wdGetter fu
...
 	if err != nil {
 		return errors.Wrapf(err, `packing of the "%v" module failed when getting commands`, moduleName)
 	}
 
-	err = packModule(loc, loc.IsDeploymentDescriptor(), module, moduleName, platform)
+	builder, _, _ := buildops.GetBuilder(module, source)
+	if builder != "zip" {
```
Comment (allaVolkov) [inline internal/artifacts/module_arch.go]: please define constant for zip builder

File: internal/artifacts/module_arch.go [hunk 5]
```diff
@@ -32,19 +32,45 @@ func ExecuteBuild(source, target, desc, moduleName, platform string, wdGetter fu
...
+// ExecuteZip - executes packing of module
+func ExecuteZip(source, target, desc, moduleName, platform string, wdGetter func() (string, error)) error {
+	logs.Logger.Infof(`zipping the "%v" module...`, moduleName)
+
+	loc, err := dir.Location(source, target, desc, wdGetter)
+	if err != nil {
+		return errors.Wrapf(err, `zipping of the "%v" module failed when initializing the location`, moduleName)
+	}
+
+	module, _, err := commands.GetModuleAndCommands(loc, source, moduleName)
```
Comment (allaVolkov) [inline internal/artifacts/module_arch.go]: don't use GetModuleAndCommands, use parse

File: internal/artifacts/module_arch.go [hunk 6]
```diff
@@ -102,6 +128,39 @@ func buildModule(mtaParser dir.IMtaParser, moduleLoc dir.IModule, deploymentDesc
...
+	}
+	// zipping the build artifacts
+	logs.Logger.Infof(`zipping the %v module...`, moduleName)
+	moduleZipFullPath := moduleZipPath + dataZip
+	sourceModuleDir := buildops.GetBuildResultsPath(ep, module)
+
+	err = dir.Archive(sourceModuleDir, moduleZipFullPath)
+	if err != nil {
+		return errors.Wrapf(err, `packing of the "%v" module failed when archiving`, moduleName)
+
```
Comment (allaVolkov) [inline internal/artifacts/module_arch.go]: remove line

File: internal/buildops/build_params.go [hunk 7]
```diff
@@ -29,13 +29,46 @@ type BuildRequires struct {
 }
 
 // GetBuilder - gets builder type of the module and indicator of custom builder
-func GetBuilder(module *mta.Module) (string, bool) {
+func GetBuilder(module *mta.Module, source string) (string, bool, map[string]string) {
```
Comment (allaVolkov) [inline internal/buildops/build_params.go]: we have to pass all needed properties for the command Zip (see related comment)

File: integration/testdata/mta_demo/node-js/server.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+/*eslint no-console: 0*/
```
Comment (ShimiT) [inline integration/testdata/mta_demo/node-js/server.js]: remove from pr

File: internal/artifacts/module_arch.go [hunk 9]
```diff
@@ -16,6 +16,11 @@ import (
 	"github.com/SAP/cloud-mta-build-tool/internal/logs"
 )
 
+const (
```
Comment (ShimiT) [inline internal/artifacts/module_arch.go]: remove

File: internal/artifacts/module_arch.go [hunk 10]
```diff
@@ -63,10 +96,10 @@ func ExecutePack(source, target, desc, moduleName, platform string, wdGetter fun
 }
 
 // buildModule - builds module
-func buildModule(mtaParser dir.IMtaParser, moduleLoc dir.IModule, deploymentDesc bool, moduleName, platform string) error {
+func buildModule(source string, mtaParser dir.IMtaParser, moduleLoc dir.IModule, deploymentDesc bool, moduleName, platform string) error {
```
Comment (ShimiT) [inline internal/artifacts/module_arch.go]: Check if the source value exist loc?

**Pull request closed without merging**